### DISCUSSION TITLE
Exporting device decoder library for use with CC1101/RFM69

### DISCUSSION THREAD

**Discussion by dwmw2** [merbanan/rtl_433#3073] [Ideas]
There are a bunch of great device decoders in rtl_433, one or two of which I've reverse-engineered contributed myself. I've had rtl_433 running on one of my tiny OpenWrt routers for years, chewing its CPU while I think that I really *ought* to replace it with something based on a CC1101 or RFM69 chip.

Would it be feasible to expose the decoders as a library which could be fed bitbuffer data from a different source? 

(See corresponding discussion in https://github.com/dbuezas/esphome-cc1101/discussions/30 )

**Comment by zuckschwerdt** [comment]
https://github.com/NorthernMan54/rtl_433_ESP is doing that :)

**Reply by dwmw2** [reply]
ooh, thanks! I'll take a look at that and see if I can build an ESPHome component out of it!
